Add pip requirements.txt and Pipfile

remingtonc opened the pull request:
Add a pip requirements.txt and Pipfile/.lock to be used for installation of the library dependencies.

This resolves the need to manually install requirements detailed in README.md and can be updated with specific versioning as required. Or updated to automatically use the latest version, just request the change in the current code.

remingtonc committed:
Merge remote-tracking branch 'OpenBMP/master'

remingtonc committed:
Add pip requirements.txt and Pipfile

Add a requirements.txt for installation via pip, as well
as Pipfile/.lock usable by Pipenv and eventual pip
Pipfile support.

TimEvens commented on Pipfile.lock:
```diff
@@ -0,0 +1,29 @@
+{
```
I was checking https://github.com/pypa/pipfile and it appears that you do not need to supply this file as it should be generated.  If it's generated, then this shouldn't be included in the repo.  Is there another reason why the lock file is included?

remingtonc replied:
The Pipfile.lock enables deterministic builds - just meaning that the **exact** same code will be utilized. We can remove this.

remingtonc replied:
http://docs.pipenv.org/en/latest/advanced.html#pipfile-lock-security-features

remingtonc replied:
https://github.com/pypa/pipfile/issues/7

Per this - we should probably leave it here unless you don't care about library versioning.

TimEvens commented on Pipfile:
```diff
@@ -0,0 +1,8 @@
+verify_ssl = true
+
+[packages]
+python-snappy = "*"
```
Can you add minimum version requirements? 

* python-snappy >= 0.5
* kafka-python >= 1.3.0
* PyYAML >= 3.11

remingtonc replied:
Should this apply to requirements.txt as well with minimum versions instead of exact (==)?

TimEvens replied:
yeah...

TimEvens commented on Pipfile:
```diff
@@ -0,0 +1,8 @@
+[[source]]
+url = "https://pypi.python.org/simple"
```
source should be: https://github.com/OpenBMP/openbmp-python-api-message

remingtonc replied:
I'm like 80% sure that the source should be the source for the project requirements - not the source of the project itself. I can't find any examples of Pipfiles that don't use pypi as a source - it might be too early in its development. https://github.com/pypa/pipfile#pipfile-1

I looked through the code for pipfile and Pipenv, and I can't tell for certain. Need to find a complete spec, which I'm not certain exists right now.

TimEvens reviewed (changes requested):
Thanks for submitting this.  I have a few comments.

remingtonc committed:
Remove pipenv files

Pipenv automatically converts requirements.txt to
Pipfile format.

remingtonc committed:
Add minimum requirements.txt versioning

remingtonc commented:
@TimEvens I have removed the Pipenv related materials and added the minimum versioning requirements to requirements.txt. Installation of dependencies is now just ```pip install -r requirements.txt```.
